src/azul/terra.py

```diff
@@ -123,6 +126,10 @@ def credentials(self) -> Credentials:
         with aws.service_account_credentials() as file_name:
             return Credentials.from_service_account_file(file_name)
 
+    @cached_property
```

Comment:
Doesn't need to be cached. The point of caching is to avoid expensive compute or resource creation. Since caching often requires trading in memory and, more importantly, consistency, it should be used with discretion.